dspace-api/src/main/java/org/dspace/app/mediafilter/PDFBoxThumbnail.java

```diff
@@ -70,10 +70,26 @@ public InputStream getDestinationStream(Item currentItem, InputStream source, bo
         throws Exception {
         BufferedImage buf;
 
+        String pageNumberStr =  currentItem.getItemService()
+            .getMetadataFirstValue(currentItem, "dspace", "filter-media", "page", Item.ANY);
+
+        int pageNumber = 0;
+
+        try {
+            if (pageNumberStr != null) {
+                pageNumber = Integer.parseInt(pageNumberStr);
+                // because of the zero base indexing
+                pageNumber -= 1;
+            }
+        } catch (Exception e) {
+            log.error("dspace.filter-media.page metadata contains an invalid page number: {}",
+                    pageNumberStr);
+        }
+
         // Render the page image.
         try ( PDDocument doc = PDDocument.load(source); ) {
             PDFRenderer renderer = new PDFRenderer(doc);
-            buf = renderer.renderImage(0);
+            buf = renderer.renderImage(pageNumber);
```

Comment:
What happens if the `pageNumber` is invalid?  For example, if we tell PDFBox to create a thumbnail from page 100 when there's only 10 pages in the document?

According to PDFBox, it appears this method will throw an IOException in that scenario.  We may want to test what occurs if this is set to an invalid number to ensure that the IOException doesn't cause additional errors / unexpected results.  It's possible we may need better error handline in this method.

Reply:
Now the handles the out of bounds exceptions in the case when the page number is more than the actual one in the file.